src/libraries/System.Net.Http.Json/src/System/Net/Http/Json/JsonContent.cs

```diff
@@ -92,7 +110,20 @@ private async Task SerializeToStreamAsyncCore(Stream targetStream, CancellationT
             }
             else
             {
-                await JsonSerializer.SerializeAsync(targetStream, Value, ObjectType, _jsonSerializerOptions, cancellationToken).ConfigureAwait(false);
+                if (async)
+                {
+                    await JsonSerializer.SerializeAsync(targetStream, Value, ObjectType, _jsonSerializerOptions, cancellationToken).ConfigureAwait(false);
+                }
+                else
+                {
+#if NETCOREAPP
+                    // Have to use Utf8JsonWriter because JsonSerializer doesn't support sync serialization into stream directly.
+                    using Utf8JsonWriter writer = new Utf8JsonWriter(targetStream);
+                    JsonSerializer.Serialize(writer, Value, ObjectType, _jsonSerializerOptions);
+#else
+                    throw new NotSupportedException(SR.SyncNotSupported);
+#endif
+                }
             }
         }
```

Comment:
Can we add tests that exercise these sync methods?

Reply:
Nevermind; I see you did below.  Do those provide coverage for all of the added overrides?

Reply:
Added extra test for `CopyTo` that hits sync `SerializeToStream`.